content/3_Silver/Binary_Search.mdx

```diff
@@ -177,33 +186,38 @@ if you're not familiar with the `Predicate` interface used.
 <PySection>
 
 ```python
-def lastTrue(lo, hi, f):
-	""" Binary Search
-	:param f: (lambda function) check a state
-	:param lo: (int) lower bound
-	:param hi: (int) upper bound
-	:return res: (int) the maximum x such that f(x) is true
+from typing import Callable
+
+
+def last_true(lo: int, hi: int, test: Callable[[int], bool]):
+	"""
+	Binary Search
+	:param lo: lower bound
+	:param hi: upper bound
+	:param test: a function that returns whether a number is valid or not
+	:return: the maximum x such that test(x) is true
 	"""
-	res = lo-1
-	while lo <= hi:
-		mid = (lo+hi)//2 # find the middle of the current range
+	lo -= 1
+	while lo < hi:
+		# find the middle of the current range (rounding up)
+		mid = lo + (hi - lo + 1) // 2
 		if f(mid):
 			# if mid works, then all numbers smaller than mid also work
-			res = mid # update the answer
-			lo = mid+1 # cut the part before mid
-			# notice that we already handled mid itself so we cut it as well
+			lo = mid
 		else:
 			# if mid does not work, greater values would not work too
-			hi = mid-1 # cut mid and after
-	# now res is the answer
-	# if res is l-1 that means no value in the range satisfies the condition
-	return res
+			# so we exclude them
+			hi = mid - 1
+	return lo
+
 
-print(lastTrue(2, 10, lambda x: True)) # 10
 # all numbers satisfy the condition
-print(lastTrue(2, 10, lambda x : x * x <= 30)) # 5
-print(lastTrue(2, 10, lambda x : False)) # 1
+print(last_true(2, 10, lambda x: True))  # outputs 10
+
+print(last_true(2, 10, lambda x : x * x <= 30))  # outputs 5
+
 # no numbers satisfy the condition
+print(last_true(2, 10, lambda x : False)) #  outputs 1
```

Comment:
be consistent with spacing (pick one of `lambda x: ` and `lambda x : `)

Reply:
oh yeah some of them might slipped my eyes